lib/factbase/lazy_taped.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025 Yegor Bugayenko
+# SPDX-License-Identifier: MIT
+
+require_relative '../factbase'
+require_relative 'taped'
+require_relative 'lazy_taped_hash'
+
+# A lazy decorator of an Array with HashMaps that defers copying until modification.
+class Factbase::LazyTaped
+  def initialize(origin)
+    @origin = origin
+    @copied = false
+    @maps = nil
+    @pairs = nil
+    @inserted = []
+    @deleted = []
+    @added = []
+  end
+
+  def pairs
+    return {} unless @pairs
+    result = {}.compare_by_identity
+    @pairs.each { |copied, original| result[copied] = original }
+    result
+  end
+
+  def inserted
+    @inserted.uniq
+  end
+
+  def deleted
```

Comment:
@HaidarJbeily7 Could you add a comment which would answer the question "why do we need this method?"